disco_aws_automation/disco_subnet.py

```diff
@@ -0,0 +1,318 @@
+                    params['RouteTableId'], params['DestinationCidrBlock'],
+                    params['VpcPeeringConnectionId'])
+                self.boto3_ec2.create_route(**params)
+            else:
+                logging.info(
+                    'update routes for (route_table: %s, dest_cidr: %s, connection: %s)',
+                    params['RouteTableId'], params['DestinationCidrBlock'],
+                    params['VpcPeeringConnectionId'])
+                self.boto3_ec2.replace_route(**params)
+
+            self._route_table = self._find_route_table()
+
+    def add_route_to_gateway(self, destination_cidr_block, gateway_id):
+        """ Try adding a route to a gateway, if fails delete matching CIDR route and try again """
+        return self._add_route(route_table_id=self.route_table['RouteTableId'],
+                               destination_cidr_block=destination_cidr_block,
+                               gateway_id=gateway_id)
+
+    def add_route_to_nat_gateway(self, destination_cidr_block, nat_gateway_id):
+        """ Try adding a route to a NAT gateway, if fails delete matching CIDR route and try again """
+        return self._add_route(route_table_id=self.route_table['RouteTableId'],
+                               destination_cidr_block=destination_cidr_block,
+                               nat_gateway_id=nat_gateway_id)
+
+    def _add_route(self, route_table_id, destination_cidr_block,
+                   gateway_id=None, instance_id=None, network_interface_id=None,
+                   vpc_peering_connection_id=None, nat_gateway_id=None):
+        params = {
+            'RouteTableId': route_table_id,
+            'DestinationCidrBlock': destination_cidr_block
+        }
+        if gateway_id:
+            params['GatewayId'] = gateway_id
+        if instance_id:
+            params['InstanceId'] = instance_id
+        if network_interface_id:
+            params['NetworkInterfaceId'] = network_interface_id
+        if vpc_peering_connection_id:
+            params['VpcPeeringConnectionId'] = vpc_peering_connection_id
+        if nat_gateway_id:
+            params['NatGatewayId'] = nat_gateway_id
+
+        result = self.boto3_ec2.create_route(**params)['Return']
+
+        if result:
+            self._route_table = self._find_route_table()
+            return result
+
+        logging.exception("Failed to create route due to conflict. Deleting old route and re-trying.")
+        delete_params = {
+            'RouteTableId': route_table_id,
+            'DestinationCidrBlock': destination_cidr_block
+        }
+        self.boto3_ec2.delete_route(**delete_params)
+
+        logging.error("Re-creating route.")
+        result = self.boto3_ec2.create_route(**params)['Return']
+        self._route_table = self._find_route_table()
+        return result
+
+    def _find_subnet(self):
+        filters = copy.deepcopy(self._resource_filter)
```

Comment:
You don't even need to copy. Use it as it is. The @property will get executed each time anyway so you'll get a new object.